volatility3/framework/symbols/linux/utilities/modules.py

```diff
@@ -204,138 +223,41 @@ def get_module_info_for_module(
 
         end = start + module.get_core_size()
 
-        return Modules.ModuleInfo(module.vol.offset, mod_name, start, end)
-
-    @staticmethod
-    def get_kernel_module_info(
-        context: interfaces.context.ContextInterface,
-        kernel_module_name: str,
-    ) -> Iterator[ModuleInfo]:
-        """
-        Returns a ModuleInfo instance that encodes the kernel
-        This is required to map function pointers to the kerenl executable
-        """
-        kernel = context.modules[kernel_module_name]
-
-        address_mask = context.layers[kernel.layer_name].address_mask
-
-        start_addr = kernel.object_from_symbol("_text")
-        start_addr = start_addr.vol.offset & address_mask
-
-        end_addr = kernel.object_from_symbol("_etext")
-        end_addr = end_addr.vol.offset & address_mask
-
-        return [
-            Modules.ModuleInfo(
-                start_addr, constants.linux.KERNEL_NAME, start_addr, end_addr
-            )
-        ]
+        return ModuleInfo(module.vol.offset, mod_name, start, end)
 
     @classmethod
-    def _get_hidden_modules_results(
-        cls,
-        context: str,
-        kernel_module_name: str,
-        run_results: Dict[str, List[ModuleInfo]],
-    ):
-        known_modules_addresses = set()
-
-        kernel = context.modules[kernel_module_name]
-
-        # Walk each sources' results
-        for results in run_results.values():
-            for modinfo in results:
-                address = context.layers[kernel.layer_name].canonicalize(modinfo.start)
-                known_modules_addresses.add(address)
-
-        modules_memory_boundaries = cls.get_modules_memory_boundaries(
-            context, kernel_module_name
-        )
-
-        hidden_results = []
-
-        address_mask = context.layers[kernel.layer_name].address_mask
-
-        for module in cls.get_hidden_modules(
-            context,
-            kernel_module_name,
-            known_modules_addresses,
-            modules_memory_boundaries,
-        ):
-            modinfo = cls.get_module_info_for_module(address_mask, module)
-            if modinfo:
-                hidden_results.append(modinfo)
-
-        return hidden_results
-
-    @classmethod
-    def _get_list_modules(
-        cls, context: interfaces.context.ContextInterface, kernel_module_name: str
-    ) -> List[ModuleInfo]:
+    def _validate_gatherers(cls, caller_wanted_gatherers) -> List[str]:
```

Comment:
Typing information on caller_wanted_gatherers please.  It must at least be `list[callable]` and then you can supply the function signature in the callable parameter if you want...

Reply:
This whole function went away just now.